File: src/com/blackoutburst/bogel/core/MouseScrollCallBack.java
package com.blackoutburst.bogel.core;

import org.lwjgl.glfw.GLFWScrollCallbackI;

/**
 * <h1>Mouse Scroll CallBack</h1>
 * 
 * <p>
 * Update mouse scroll <b>do not call</b>
 * </p>
 * 
 * @since 0.1
 * @author Blackoutburst
 */
public class MouseScrollCallBack implements GLFWScrollCallbackI {

	@Override
	public void invoke(long window, double xOffset, double yOffset) {
		Mouse.setScroll((float) (Mouse.getScroll() + yOffset));
	}
}

File: client/src/main/kotlin/dev/blackoutburst/game/utils/MouseScrollCallBack.kt
package dev.blackoutburst.game.utils

import org.lwjgl.glfw.GLFWScrollCallbackI

class MouseScrollCallBack : GLFWScrollCallbackI {
    override fun invoke(window: Long, xOffset: Double, yOffset: Double) {
        Mouse.scroll = (Mouse.scroll + yOffset).toFloat()
    }
}
